### PR TITLE
enhancement: Verbose Test Harness

### PR DESCRIPTION
**Pull request by tzaffi**
## Summary

To improve visibility of C.I. build failures, utilize new logging capabilities in the sandbox (https://github.com/algorand/sandbox/pull/181) and set `VERBOSE_HARNESS=1`.

In particular we can now see:
* traces of the `docker compose up` command
* dumped `docker-compose logs` for:
  * indexer-db
  * algod
  * indexer

### TODO

- [x] open similar PR's in the other SDK's
- [x] Wait for #538 to get merged
- [x] Wait for https://github.com/algorand/algorand-sdk-testing/pull/288 to get merged

### What does a successful build look like?
See for example [this build](https://app.circleci.com/pipelines/github/algorand/go-algorand-sdk/954/workflows/bc9b8ca1-7ed0-43a4-8b3b-fbc790e2b939/jobs/2309?invite=true#step-108-1). You'll need to download Circle's log as the max display at Circle is 400K characters.

### What does a failed build look like?
As a test case I pinned the companion algorand-sdk-testing PR (https://github.com/algorand/algorand-sdk-testing/pull/288) to a [broken indexer commit](https://github.com/algorand/algorand-sdk-testing/pull/288/files/53fc00465299383ca19984c6c8f7f04d25556157..b7f0e7ae3c66db6e00a423974aaad0cd0ba4fb88). The build broke early enough to see the entire logs directly on circle. There are two highlights to look for:
1. We can see the usual error information [one or more services failed to start](https://app.circleci.com/pipelines/github/algorand/go-algorand-sdk/964/workflows/4c4b6d38-4b6a-484b-ae90-90395c90de0c/jobs/2319?invite=true#step-108-3172)
2. Then the `docker compose logs` are dumped for each of the services and at the end we see [could not load genesis...](https://app.circleci.com/pipelines/github/algorand/go-algorand-sdk/964/workflows/4c4b6d38-4b6a-484b-ae90-90395c90de0c/jobs/2319?invite=true#step-108-3172)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,6 +39,6 @@ We use golangci-lint to run linters on our codebase. Please run `make lint` befo
 
 To download the SDK, open a terminal and use the `go get` command.
 
-```command
+```sh
```
Comment (tzaffi) [inline README.md:42]:
```suggestion
```command
```

Comment (tzaffi) [inline README.md:42]:
```suggestion
```command
```

Comment (tzaffi) [reply README.md:42]: trigger buidl

File: .test-env [hunk 2]
```diff
@@ -1,11 +1,11 @@
 # Configs for testing repo download:
 SDK_TESTING_URL="https://github.com/algorand/algorand-sdk-testing"
-SDK_TESTING_BRANCH="pre-consensus"
+SDK_TESTING_BRANCH="verbose-harness"
 SDK_TESTING_HARNESS="test-harness"
 
 INSTALL_ONLY=0
 
-VERBOSE_HARNESS=0
+VERBOSE_HARNESS=1
```
Comment (tzaffi) [inline .test-env:8]: Once the `sandbox` and `sdk-testing` PR's are merged, this is the only change that is needed in the other SDK's

File: README.md (hunk 1 above)
Comment (tzaffi) [inline README.md:42]:
```suggestion
```command
```
wait for the build to pass

File: README.md [hunk 3]
```diff
@@ -9,13 +9,13 @@ The Algorand golang SDK provides:
...
 
-# Documentation
+## Documentation
 
 Full documentation is available [on pkg.go.dev](https://pkg.go.dev/github.com/algorand/go-algorand-sdk/v2). You can also self-host the documentation by running `godoc -http=:8099` and visiting `http://localhost:8099/pkg/github.com/algorand/go-algorand-sdk` in your web browser.
 
 Additional developer documentation and examples can be found on [developer.algorand.org](https://developer.algorand.org/docs/sdks/go/)
 
-# Package overview
+## Package overview
```
Comment (tzaffi) [inline README.md:18]:
```suggestion
## Package Overview
```

File: .test-env (hunk 2 above)
Comment (algochoi) [reply .test-env:8]: LGTM - do we want to target the `develop` branch instead or roll in this change with this branch?

File: .test-env [hunk 4]
```diff
@@ -1,11 +1,11 @@
 # Configs for testing repo download:
 SDK_TESTING_URL="https://github.com/algorand/algorand-sdk-testing"
-SDK_TESTING_BRANCH="pre-consensus"
+SDK_TESTING_BRANCH="verbose-harness"
```
Comment (tzaffi) [inline .test-env:3]:
```suggestion
SDK_TESTING_BRANCH="master"
```
TODO: don't merge without reverting

File: .test-env (hunk 2 above)
Comment (tzaffi) [reply .test-env:8]: But if you prefer to roll it into `txn-deltas-api` that works as well.

Comment (algochoi) [reply .test-env:8]: I think targeting `develop` is fine since indexer is building on sandbox again!

**Review by Eric-Warehime — COMMENTED** [review]
LGTM, you just need to update the sdk testing branch to master

**Review by winder — APPROVED** [review]
LGTM, just need to update the branch.

**Review by algochoi — APPROVED** [review]
LGTM, thanks! very helpful for debugging

**Pull request merged**